### PR TITLE
TrackletEventProcessor: properly initialise and check settings

### PR DESCRIPTION
**Pull request by fwyzard**
Initialise `settings` to `nullptr` in the constructor.
Check that `settings` is not `nullptr` before accessing it in the destructor.
Pass `settings` by reference to avoid the possibility of a null pointer.

### CODE REVIEW

**Comment by fwyzard** [comment]
type bug-fx

**Comment by fwyzard** [comment]
urgent

**Comment by fwyzard** [comment]
backport #30659

**Comment by fwyzard** [comment]
@cmsbuild please test

**Comment by silviodonato** [comment]
merge

**Comment by rekovic** [comment]
+1

File: L1Trigger/TrackFindingTracklet/src/TrackletEventProcessor.cc [hunk 1]
```diff
@@ -14,16 +14,16 @@
...
-  if (settings_->bookHistos()) {
+  if (settings_ && settings_->bookHistos()) {
     histbase_->close();
   }
 }
 
-void TrackletEventProcessor::init(const Settings* theSettings) {
-  settings_ = theSettings;
+void TrackletEventProcessor::init(Settings const& theSettings) {
+  settings_ = &theSettings;
```
Comment (davidlange6) [inline L1Trigger/TrackFindingTracklet/src/TrackletEventProcessor.cc:26]: the issue is not created by this PR - however there is no promise that the memory allocated to theSettings will stay around for the lifetime of TrackletEventProcessor (which this construct depends on)

**Pull request merged**